tests/playwright/sanity/core/base/document.test.ts

```diff
@@ -51,7 +51,13 @@ test.describe( 'Document tests', async () => {
 
 async function addElement( wpAdmin: WpAdminPage, elementType: string ) {
 	const frame = wpAdmin.page.frame( { name: 'editor-canvas' } );
-	await frame.click( '.block-editor-inserter__toggle' );
+	if ( ! await wpAdmin.page.frameLocator( 'iframe[name="editor-canvas"]' ).locator( 'p[role="document"]' ).isVisible() ) {
```

Comment:
How about doing something like this once:
```typescript
await wait wpAdmin.page.frameLocator( 'iframe[name="editor-canvas"]' ).addLocatorHandler('p[role="document"]', async overlay => {
  await frame.locator( '.block-editor-inserter__toggle' ).click();
}, { times: 3, noWaitAfter: true });
```

Reply:
To be clear - this needs to be done after page load, just once, not here

Reply:
see: https://playwright.dev/docs/api/class-page#page-add-locator-handler

Reply:
I'm not sure I understand the logic. You want to click + button 3 times?
The problem here is, that first time Playwright clicks + button, it is in iframe. Next + button is not.

Reply:
> see: https://playwright.dev/docs/api/class-page#page-add-locator-handler

I'm not sure this is a case for that. Here we have a clear case. If no element was added- + in Iframe, if at least one is there, then it is not in iframe

Reply:
I want that after the iframe is rendered to add a locator handler that will make these popups go away, regardless of what action it is we perform